crates/volta-core/src/command.rs

```diff
@@ -28,3 +31,52 @@ cfg_if! {
         }
     }
 }
+
+/// Rebuild command against given PATH
+///
+/// On Windows, we need to explicitly use an absolute path to the executable,
+/// otherwise the executable will not be located properly, even if we've set the PATH.
+/// see: https://github.com/rust-lang/rust/issues/37519
+///
+/// This function will try to find the executable in the given path and rebuild
+/// the command with the absolute path to the executable.
+pub fn rebuild_command<S: AsRef<OsStr>>(command: Command, path: S) -> Fallible<Command> {
+    debug!("PATH: {}", path.as_ref().to_string_lossy());
+
+    #[cfg(not(windows))]
+    {
+        let mut command = command;
+        command.env("PATH", path.as_ref());
+        Ok(command)
+    }
+
+    #[cfg(windows)]
+    {
+        let args = command.get_args().collect::<Vec<_>>();
+        //          cmd /c <name> [...other]
+        // args_idx     0  1      2..
+        let name = args.get(1).expect("should have the name");
+
+        if let Ok(mut paths) = which::which_in_global(name, Some(&path)) {
+            if let Some(exe) = paths.next() {
```

Comment:
As a stylistic thing (not a blocker!) since we're on a version of Rust which supports it, I prefer the `let … else` form, which flattens the code flow out a lot. It *does* require us to write the error out twice in that case, though, so it’s a small tradeoff around readability!